geonode/security/models.py

```diff
@@ -135,6 +135,10 @@ def set_default_permissions(self):
 
         # default permissions for anonymous users
         anonymous_group, created = Group.objects.get_or_create(name='anonymous')
+
+        if not anonymous_group:
```

Comment:
What is the case when get_or_create fails? and isn't it raising an exception in case of fail?